libguard/guard_interface.cpp

```diff
@@ -167,9 +171,20 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType)
             }
             return getHostEndiannessRecord(existGuard);
         }
+
+        // Storing the 1st resolved guard record position
+        if (existGuard.recordId == GUARD_RESOLVED)
+        {
+            if (!isResolvedRecordPresent)
```

Comment:
Cannot merge the inner `if` block with the outer `if` block?

Reply:
Yes we can merge it.